src/main/java/com/sap/sgs/phosphor/fosstars/model/score/FeatureBasedScore.java

```diff
@@ -50,13 +51,18 @@ public FeatureBasedScore(String name, String description, Feature... features) {
    */
   public FeatureBasedScore(String name, String description, Set<Feature> features) {
     super(name, description);
-    this.features = Collections.unmodifiableSet(features);
+    for (Feature feature : features) {
+      if (feature instanceof Score) {
+        throw new IllegalArgumentException("Hey! I cannot work with scores!");
+      }
+    }
+    this.features = new HashSet<>(features);
   }
 
   @Override
   @JsonIgnore
   public final Set<Feature> features() {
-    return features;
+    return new HashSet<>(features);
```

Comment:
The features is already set to `new HashSet<>()` in constructor L59.
We can just return the `features` right?

Reply:
If the method returns the `features` field, then the caller can modify the field. If the constructor doesn't wrap feature into a new set, then the caller can modify the internal `features` field as well. That's why wrapping is necessary in both places.